deploy.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+help() {
+    cat <<EOH
+    Usage: $0 [OPTIONS] <ENVIRONMENT>
+
+    Options:
+      -h        Show this help.
+      -c        Force refresh the certificates (only for new installations or
+                if they are expired).
+      -b        Also build the container image (locally only).
+      -v        Show verbose output.
+
+EOH
+}
+
+check_environment() {
+    # verify that the proper environment is passed
+
+    if [[ ! -d "deploy/$environment" || "$environment" =~ ^(base)$ ]]; then
+        echo "Unknown environment $environment, use one of:"
+        ls deploy/ | egrep -v '^(base)'
+        exit 1
+    fi
+}
+
+build_image() {
+    # build container image for dev or prod environments
+
+    local environment="${1?No environment passed}"
+    local do_build="${2?No build argument passed}"
+
+    if [[ "$do_build" == "yes" ]]; then
+        if [[ "$environment" == "devel" ]]
+        then
+            # creates the image on minikube's docker daemon
+            eval $(minikube docker-env)
+            docker build -t buildpack-admission:latest .
+        else
+            # Build the container image on the docker-builder host (currently tools-docker-imagebuilder-01.tools.eqiad1.wikimedia.cloud).
+            docker build . -f Dockerfile -t docker-registry.tools.wmflabs.org/buildpack-admission:latest
+            # Push the image to the internal repo
+            docker push docker-registry.tools.wmflabs.org/buildpack-admission:latest
+            echo "Successfully built container image for toolforge/toolsbeta environments with exit code 0. \
+            To deploy,log into k8s control node with repository checked out and run './deploy.sh (toolforge or toolsbeta)'"
+            exit 0
+        fi
+    fi
+}
+
+regenerate_certs() {
+    # creates a new certificate, a CSR to sign it, and a k8s secret with the signed cert and key
+
+    local refresh_certs="${1?No refresh_certs argument passed}"
+    if [[ "$refresh_certs" == "yes" ]]; then
+        ./utils/regenerate_certs.sh
+    fi
+}
+
+deploy_generic() {
+    # deploy buildpack-admission-controller image to either dev or prod environments
+
+    local environment="${1?No environment passed}"
+    local refresh_certs="${2?No refresh_certs argument passed}"
+
+    if [[ "$environment" == "devel" ]]
+    then
+        # generate the patch to override the ca bundle with the k8s secret we just created
+        ./utils/realize_patch.sh deploy/devel/webhook.patch.yaml.tpl
+        # Deploy the dev environment
+        kubectl apply -k deploy/devel
+    else
+        if [[ "$refresh_certs" == "yes" ]]
+        then
+            # do rolling restart to ensure the new certs created by regenerate_certs is being used
+            kubectl rollout restart -n buildpack-admission deployment/buildpack-admission
+        else
+            kubectl --as=admin --as-group=system:master apply -k deploy/toolforge
```

Comment:
this should use the $environment, instead of hardcodding toolforge

Reply:
> s/system:master/system:masters/

I don't think I understand what you meant here @david-caro

Reply:
> > s/system:master/system:masters/
> 
> I don't think I understand what you meant here @david-caro

Oh sorry!

That comments means substitute `system:master` for `system:masters`

That is a comment that comes from overusing vim and similar others, where you can specify a command to do the substitution (ex https://vim.fandom.com/wiki/Search_and_replace).

Reply:
> > > s/system:master/system:masters/
> > 
> > 
> > I don't think I understand what you meant here @david-caro
> 
> Oh sorry!
> 
> That comments means substitute `system:master` for `system:masters`
> 
> That is a comment that comes from overusing vim and similar others, where you can specify a command to do the substitution (ex https://vim.fandom.com/wiki/Search_and_replace).

ooo interesting! I've never used vim so it was confusing, thanks for pointing it out